### ISSUE TITLE
AutoDraw Roadmap

### ISSUE THREAD

**Issue by Siydge** [auto-draw/autodraw#5]
This issue is to track the Roadmap changes for AutoDraw, to list all features planned or complete for the future Milestone updates. This issue is open to discussion for newer suggestions, however use of the Discord Server for suggestions is preferred.

### TODO:
---
- [ ] Image Directory & Queuing: 
Store and Load multiple images, or a set of images with same or differing settings, alongside a set actions to perform before or after drawing, to automate drawing sequences.
The set of actions must support the input of specified text along with mouse movements.
- [x] Improved Drawing Algorithm: 
Add a revamped drawing algorithm that will follow an optimal path, based off of contour filling paths, also starting from largest chunks to smallest.
- [x] Save Locked Position: 
Save the last used position and settings for a specified image, in case of accidental closure or a crash occurs, so that manual realignment and inputting previous values isn't required.
- [ ] Improve Image Settings: 
Allow Input of Minimum or Maximum resolutions for images, so that all images follow those limits.
- [x] Improve Image Viewer:
Allow Zoom and Panning on Image
Show Color Details of Hovered Pixel (Hex, RGB, HSL, etc)
Show Other Additional Details of Image & Hovered Pixel.
- [ ] Smart Click Delay:
Basically runs Click Delay only when previous click was recent enough
- [ ] Improve Image Editing:
Add support for Rotating Images
Add support for Cropping Images
Allow changing scale modes
- [ ] Full Linux Support
Add support for AutoDraw's native menu bar 
Add support for mouse movement in both X11 and Wayland
Add support for drag/drop and pasting

**Possible Additions**
- [ ] Basic/Lightweight Color Splitter
Could be added alongside the Image Directory & Queuing
Will not have fine tuning or anything excessive, for more in-depth or precise control, that will be available with different planned program, separate from AutoDraw

**Comment by MrWaffleManTheGreat** [comment]
Amazing Work So Far!, I have a idea that will help the fuzzyness of drawings. 

Border Drawer:
A feature that (will) draw around the border of the drawing in the direction of the nearest strokes after finishing a drawing. this will remove the fuzzyness around the edges cause by the directional drawing.

**Comment by Siydge** [comment]
> Border Drawer: A feature that (will) draw around the border of the drawing in the direction of the nearest strokes after finishing a drawing. this will remove the fuzzyness around the edges cause by the directional drawing.

The plan is for the drawing algorithm to completely remove directional drawing. It'll instead draw in pattern of the shape, so if its a filled in circle, it'll draw it as a circle, drawing inwards. in a circular motion. That way images will look normal and should reduce issues.

> Circlular Drawing Pattern:
> Makes small circles instead of the directional horizontal/vertical drawing pattern

I may do this for small points and dots if possible, but thats a strong maybe as in some cases we're working in 1x1 pixels, and I still want to ensure a close 1:1 replica of the image input.

**Comment by Itsukiomo** [comment]
Are you still working on this??

**Comment by Itsukiomo** [comment]
Oh I have an idea if theres a way for u to make this draw with the color of the image. You make it so people can set up what to press to switch color so it works on pretty much any game. For example I can open the color picker with F and map each color on the software where the color is in game. Kinda like a macro. You can add 20-40 different colors that its supports.

**Comment by DaltoGit** [comment]
Is this project still being worked on?

**Comment by Siydge** [comment]
Development has halted for v2, since I'd like to move to Rust for proper Cross-Platform support (Linux, Windows, MacOS) and the libraries on Rust are more capable, on top of Avalonia having more slow/sluggish loading than I'd like.
I haven't had much energy to work on the Rust rewrite unfortunately.
